### PR TITLE
Raise DARHTTPException also for retried status codes

### LINKED ISSUE #104: RetryException not handled as HTTP-related error
Requests can internally raise a RetryError:

```
requests.exceptions.RetryError: HTTPSConnectionPool(host='XXX.sap.hana.ondemand.com', port=443): Max retries exceeded with url: /data-manager/api/v3/datasets (Caused by ResponseError('too many 500 error responses'))
```

This should be handled similar to the existing handling for HTTP error codes; i.e. raise an exception which includes request details.

### PR DESCRIPTION
**Pull request by mhaas**
For retried HTTP status codes such as 500, the `requests` library would
internally raise a `RetryError`. As a consequence, the `DARSession`
did not raise the `DARHTTPException` which contains important
debugging information.

This commit fixes issue #104 by configuring `requests` (rather, `urllib3`)
to return the last response instead of raising an exception. The response
contains the HTTP 500 status code which is then converted to a `DARHTTPException`
in the `DARSession`.

The initial approach to handle this issue was to catch the `RetryException` and
extract the last `Response` object from the exception. However, this is not possible
due to a limitation in `urllib3`: https://github.com/urllib3/urllib3/issues/2096

This will make debugging HTTP 500 issues much easier.

Fixes #104

- [x] Maintain changelog

### CODE REVIEW

File: tests/sap/aibus/dar/client/test_dar_session.py [hunk 1]
```diff
@@ -251,3 +253,88 @@ def test_plain_http_is_not_allowed(self):
...
+    return "https://localhost/"
+
+
+@pytest.fixture
+def credentials_source():
+    return StaticCredentialsSource("12345")
+
+
+@pytest.fixture
+def dar_session_500_error(dar_url, credentials_source):
```
Comment (mhaas) [inline tests/sap/aibus/dar/client/test_dar_session.py:269]: Not sure if you are familiar with this syntax: pytest will automatically provide the output of the fixtures in line 258 and 263.

Comment (karthikmswamy) [reply tests/sap/aibus/dar/client/test_dar_session.py:269]: Thanks for the info

**Review by karthikmswamy — APPROVED** [review]
Thank you! Looks good to me.

**Comment by karthikmswamy** [comment]
@mhaas, quick reminder to update changelog before merging this PR.

**Pull request merged**